Sources/GRPCCore/Metadata.swift

```diff
@@ -494,15 +494,23 @@ extension Metadata.Value: ExpressibleByArrayLiteral {
 
 extension Metadata: CustomStringConvertible {
   public var description: String {
-    String(describing: self.map({ ($0.key, $0.value) }))
+    var description = "["
+    description = self.dropLast().reduce(into: description) { partialResult, nextPair in
+      partialResult += "\"\(nextPair.key)\": \(nextPair.value), "
+    }
+    if let lastElement = self.last {
+      description += "\"\(lastElement.key)\": \(lastElement.value)"
+    }
+    description += "]"
+    return description
   }
 }
 
 extension Metadata.Value: CustomStringConvertible {
   public var description: String {
     switch self {
     case .string(let stringValue):
-      return String(describing: stringValue)
+      return "\"\(stringValue)\""
```

Comment:
This has the downside that when printing an individual value it will be quoted. When you print a sequence the `CustomDebugStringConvertible` conformance will be used. This is why `print(["foo"])` prints `["foo"]` and `print("foo")` prints `foo`. We should use `CustomDebugStringConvertible` here.

Reply:
I think it's obvious without quotes for a single value: `[0, 1, 2]` vs `foo`.

Reply:
Well, it could be the string "[1, 2, 3]" but maybe I'm too cynical